src/backend/distributed/planner/function_call_delegation.c

```diff
@@ -112,9 +112,15 @@ TryToDelegateFunctionCall(DistributedPlanningContext *planContext)
 	}
 
 	int32 localGroupId = GetLocalGroupId();
-	if (localGroupId != COORDINATOR_GROUP_ID || localGroupId == GROUP_ID_UPGRADING)
+	if (localGroupId != COORDINATOR_GROUP_ID && IsCitusInitiatedRemoteBackend())
```

Comment:
minor: can you point me to the regression test that works fine because we have this change? Also, we could use more elaborate comment here (with better style and :
```
/* do not delegate from workers if it is initiated by Citus already. It means that this function has already been delegated to this node */
```
Also, please update the comment for procedure delegation as well

Reply:
I've updated the comment and you can find the effected test from https://github.com/citusdata/citus/pull/5415/files#diff-92f6fca4762115d449d76306c6b5cc379f7d836ad43d62c08623f0ae73707a72R725

Reply:
That is a test for colocated to distributed function pushdown?
```SQL
-- Mark them as colocated with a table. Now we should route them to workers.
select colocate_proc_with_table('mx_call_func', 'mx_call_dist_table_1'::regclass, 1);

```

I thought this was for reference tables with task-assignment-policy:round-robin. 

Because for colocated-to-distributed-table functions are already handled with `ereport(DEBUG1, (errmsg("not pushing down function to the same node")));` check as they are already routed to the correct node. 

But, for reference tables, we can keep routing the same function until `ereport(DEBUG1, (errmsg("not pushing down function to the same node")));` check, where with this one we want to push only once.

Is my understanding wrong?

Reply:
For reference table related one, you can check the https://github.com/citusdata/citus/blob/6feb0098343b474ebad06a1c28321c944555095c/src/test/regress/expected/multi_mx_function_table_reference.out#L154